[Warlock] Move Havoc handling to main module for Covenant usage

Azevara opened the pull request.

Azevara committed:
Move Havoc handling to main module for Covenant usage

nuoHep committed:
Spell Info: print all affecting spell effect indices

nuoHep committed:
Update spell data dump

Hinalover committed:
[Monk] Convert Breath of Fire to an Active Action
Considering there are 3 different places that trigger the Breath of Fire dot, converting to an active action makes things easier to trigger the dot then separate spells for each.

- update strength/agility to parry/dodge conversion for levels 1-12

seanpeters86 committed:
[Priest] dots auto benefit from mastery (#5309)

* [Priest] dots auto benefit from mastery

* fix issues from review

* fix error from refactor

seanpeters86 committed:
[Priest] whitelist fae fermata buff

gastank committed:
[Guardian] the natural order's will initial implementation

gastank committed:
[Druid] fix convoke the spirits to ignore last partial tick

EvanMichaels committed:
[Rogue] Minor Poison/Blade Flurry Refactor

* Add procs_blade_flurry() function for better controlling Blade Flurry instead of hard-coding exceptions
* Clean up Killing spree string comparisons in favor of ID checks
* Move poison triggers to schedule_travel based on in-game testing
* Set poisons to not trigger Blade Flurry
* Swap Fan of Knives to use generic procs_poison() functionality
* Flag Shadow Vault and Shuriken Storm as being able to proc poisons

gastank committed:
[Druid] enable eclipse only for balance or balance affinity

Hinalover committed:
[Monk] No longer hard-code Blackout Kick's Weapons of Order Cooldown Reduction

vituscze committed:
Add dot_t::ticks_left_fractional

gastank committed:
[Druid] add `lycaras_fleeting_glimpse` 5s warning buff

seanpeters86 committed:
[Priest] add bugged behavior for DP refresh

vituscze commented:
`destruction_spell_t::init()` should also be moved, otherwise you risk having spells with `use_havoc` which are not in the `havoc_spells` vector.

vituscze committed:
Guard against buffs skipping max_stack() - offset stack

vituscze committed:
[Mage] Simplify some max_stack() expressions

gastank committed:
[Druid] disable kindred empowerments partner damage for guardian/resto

gastank committed:
[Druid] add helper to convoke_the_spirits to handle dynamic_cast for free_cast

Bloodmallet committed:
update PreRaid profile restrictions to sane values

Discussion will need to happen, this is just a sanity increase

scamille committed:
Update profiles (#5318)

* Update T25 profiles

* Add working profiles to CI.simc

* Disable warriors

vituscze reviewed (changes requested):
`destruction_spell_t::init()` should also be moved, otherwise, you risk having spells with `use_havoc` which are not in the `havoc_spells` vector.

vituscze committed:
Adjust dot_t::at_max_stacks as well

gastank committed:
[Gear] empty framework for generic runecarves

seanpeters86 committed:
[Priest] add clarifying comment on bug impl

gastank committed:
[Druid] remove spec/affinty check for parsing `eclipse.X` expr

gastank committed:
Add interrupt_proc_type to action_state debug message

gastank committed:
[Druid] set interrupts as harmful=false

gastank committed:
[Gear] sephuz's proclamation initial implementation

gastank committed:
[Soulbind] use at_max_stacks() for counter buffs

gastank committed:
[Gear] norgannon's sagacity initial implementation

gastank committed:
Default to -8 scaling for player scaling effects with a max scaling level

For spells that...
    a. have no explicit scaling class set in the data
    b. are not flagged to scale with item level
    c. have a maximum scaling level set in the data

... their in-game values match -8 scaling, so spelleffect_data_t::average( player_t ) will apply -8 scaling class if the above three conditions are met.

gastank committed:
apply 15ae2bab80 fix to CLI spell_query output

Azevara committed:
Merge branch 'move_havoc' of https://github.com/Azevara/simc into move_havoc
trying to finish havoc shit

Azevara committed:
Move init() code

Azevara commented:
Easier to close and resubmit cleanly because man did i screw this up
